src/access/AccessControl.cpp

```diff
@@ -115,12 +117,45 @@ CHIP_ERROR AccessControl::Check(const SubjectDescriptor & subjectDescriptor, con
             {
                 NodeId subject = kUndefinedNodeId;
                 ReturnErrorOnFailure(entry.GetSubject(i, subject));
-                if (subject == subjectDescriptor.subjects[0])
+                if (IsOperationalNodeId(subject))
                 {
-                    subjectMatched = true;
-                    break;
+                    if (subject == subjectDescriptor.subject)
+                    {
+                        subjectMatched = true;
+                        break;
+                    }
+                }
+                else if (IsGroupId(subject))
+                {
+                    VerifyOrReturnError(authMode == AuthMode::kGroup, CHIP_ERROR_INVALID_ARGUMENT);
+                    if (subject == subjectDescriptor.subject)
+                    {
+                        subjectMatched = true;
+                        break;
+                    }
+                }
+                else if (IsPAKEKeyId(subject))
+                {
+                    VerifyOrReturnError(authMode == AuthMode::kPase, CHIP_ERROR_INVALID_ARGUMENT);
+                    if (subject == subjectDescriptor.subject)
+                    {
+                        subjectMatched = true;
+                        break;
+                    }
+                }
+                else if (IsCASEAuthTag(subject))
+                {
+                    VerifyOrReturnError(authMode == AuthMode::kCase, CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
Why not continue here, and in all other similar subcases?

Reply:
This is invalid ACL entry, I guess we should signal an error in this case and prevent any use of invalid entries. Are you proposing just ignore invalid entry?

Reply:
I consider this to be fatal error. When I add all the error-checking code where it's being set, this will never happen, because we would not have created or updated the entry in the first place. If it ever were to happen, sorry you will need factory reset. If we had a mysterious bad entry (e.g. some non-access-control code overwrote the flash?) why would we trust other entries?